pkg/supervisor/mock.go

```diff
@@ -18,26 +18,21 @@
 package supervisor
 
 import (
-	"sync"
-
 	"github.com/megaease/easegress/pkg/cluster"
 	"github.com/megaease/easegress/pkg/option"
 )
 
 // NewMock return a mock supervisor for testing purpose
-func NewMock(options *option.Options, cls cluster.Cluster, businessControllers sync.Map,
-	systemControllers sync.Map, objectRegistry *ObjectRegistry, watcher *ObjectEntityWatcher,
-	firstHandle bool, firstHandleDone chan struct{}, done chan struct{}) *Supervisor {
+func NewMock(options *option.Options, cls cluster.Cluster, objectRegistry *ObjectRegistry, watcher *ObjectEntityWatcher,
+	firstHandle bool, firstHandleDone, done chan struct{}) *Supervisor {
 	return &Supervisor{
-		options:             options,
-		cls:                 cls,
-		businessControllers: businessControllers,
-		systemControllers:   systemControllers,
-		objectRegistry:      objectRegistry,
-		watcher:             watcher,
-		firstHandle:         firstHandle,
-		firstHandleDone:     firstHandleDone,
-		done:                done,
+		options:         options,
+		cls:             cls,
+		objectRegistry:  objectRegistry,
+		watcher:         watcher,
+		firstHandle:     firstHandle,
+		firstHandleDone: firstHandleDone,
+		done:            done,
```

Comment:
I am not sure it is good or not to remove `businessController` and `systemControllers`?

Reply:
Actually, there is a problem that sync.Map contains sync.Mutex, it's pass-by-value.

Reply:
`passed-by-value` or `passed-by-reference` is a choice in a different scenario.  As this is a New Object function. pass-by-value might make sense.  @localvar please confirm this is a problem to not.

But what I am concerned about, these changes totally modified the function signature, and the reason is we need to align with a lint tool instead of trying to optimize the function interface and make it more usable or readable.

Reply:
> But what I am concerned about, these changes totally modified the function signature, and the reason is we need to align with a lint tool instead of trying to optimize the function interface and make it more usable or readable.
It's configurable.

Reply:
`sync.Map` contains `sync.Mutex` and `atomic.Value`, both "must not be copied after first use", it is not a problem in this case because they are not used yet.

But I think we should avoid this `passed-by-value` because it is hard to track whether they are used or not in all cases, which makes the code error-prone.

Reply:
I think this is a problem even they are unused. at least, it is a potential problem, which must be fixed.

The second question is - whether are those two fields are useless? if it is, we should remove it, otherwise, we need to fix it.

Reply:
They have usages in production code. Because function `NewMock` is for testing, we can remove the assignment from `NewMock` by now (and add them back if they are required later, in a manner other than `passed-by-value`), as what @kevwan do in this PR.

Reply:
I ignored it by new commits. Because there are a few more issues like this. I'll might work on all of them later.

Let's just keep this PR simple enough.